src/components/BookList.js

```diff
@@ -1,25 +1,55 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { useState } from 'react';
 import Book from './Book';
+import { addBook } from '../redux/features/books/booksSlice';
 
 const BookList = () => {
-  const books = [
-    { id: '1', title: 'The Hunger games', author: 'Suzanne Collins' },
-    { id: '2', title: 'Dune', author: 'Frank Herbert' },
-    { id: '3', title: 'Capital in the Twenty-First Century', author: 'Suzanne Collins' },
-  ];
+  const { books } = useSelector((store) => store.books);
+  const dispatch = useDispatch();
+  const [newBookInfo, setNewBookInfo] = useState({
+    title: '',
+    author: '',
+  });
+
+  const handleInputs = (e) => {
+    if (e.target.name === 'title') {
+      setNewBookInfo((prevState) => ({
+        ...prevState,
+        title: e.target.value,
+      }));
+    } else if (e.target.name === 'author') {
+      setNewBookInfo((prevState) => ({
+        ...prevState,
+        author: e.target.value,
+      }));
+    }
+  };
+
+  const addNewBook = (e) => {
+    e.preventDefault();
+    const id = Math.floor(Math.random() * 100000);
+    if (newBookInfo.title.trim() && newBookInfo.author.trim()) {
+      dispatch(addBook({
+        item_id: id.toString(),
+        title: newBookInfo.title,
+        author: newBookInfo.author,
+      }));
+    }
+  };
```

Comment:
- [OPTIONAL] After submitting the form data for each book, it would be nice to clear the data. It would improve the user experience. 👍 😎

Reply:
I will implement it ASAP, thanks @Gambit142  🙏